### PR TITLE
separate hubs from similar games

### PR DESCRIPTION
**Pull request by Jamiras**
Extracts Hubs from the Related list and turns them into metadata which will be displayed below the game screenshots.

image image

If a Hub name starts with Meta, that's discarded. The remaining Hub name is split on the last hyphen, creating a label and value. Multiple items sharing a label are grouped together, otherwise labels are ordered as they would have been in the Related list.

The metadata items appearing above the screenshots are still populated from the game details (editable in the Dev section), but matching Hub items will be merged. If the label/values are not exact, they will be appended instead of merged:

image

### CODE REVIEW

File: public/gameInfo.php [hunk 1]
```diff
@@ -560,36 +569,57 @@ function submitSetRequest(user, gameID) {
...
+
+                $first = true;
+                if ($gameHubs) {
+                    $hubPrefix = "[$label - ";
+                    foreach ($gameHubs as $hub) {
+                        if (substr($hub['Title'], 0, strlen($hubPrefix)) == $hubPrefix) {
+                            if ($first) {
+                                $first = false;
+                            } else {
+                                $output .= ", ";
```
Comment (luchaos) [inline public/gameInfo.php:584]: Minor thing - alternatively, I'd probably collect all the outputs in an array and implode it with a separator after the loop.
Then there's no need for the local `$first` variable and the `if/else` construct.
```
$hubLinks = [];
foreach ($gameHubs as $hub) {
    $hubLinks[] = "<a href=/game/" . $hub['gameIDAlt'] . ">$value</a>";
}
$output .= implode(', ', $hubLinks);
```

Not sure how it affects `$value == $gameDataValue` and the other check below - those might be reduced to filtering the array of hubs beforehand?

File: public/gameInfo.php [hunk 2]
```diff
@@ -560,36 +569,57 @@ function submitSetRequest(user, gameID) {
             $imageIngame = $gameData['ImageIngame'];
             $pageTitleAttr = attributeEscape($pageTitle);
 
+            function addMetadata($label, $gameDataValue, $gameHubs)
```
Comment (luchaos) [inline public/gameInfo.php:572]: As this function renders HTML I'd probably put it in the render lib folder - even though this application is only used here atm. Yet, I don't think we keep functions within "endpoint" files anywhere else.
I'd suggest `renderMetadata()` for the function name as it does in fact render, not add metadata?
That'd be mostly in line with the rest of the application, too.

**Review by luchaos — COMMENTED** [review]
Wonderful idea to add that data to the game details. Some minor code calisthenics I'd personally change - feel free to discard though, nothing important there. It works fine.

**Review by luchaos — APPROVED** [review]

File: public/gameInfo.php (hunk 1 above)
Comment (Jamiras) [reply public/gameInfo.php]: Good idea. I forget about language-specific tricks like this.

File: public/gameInfo.php (hunk 2 above)
Comment (Jamiras) [reply public/gameInfo.php]: The function was created to reuse code. It's kind of awkward to put that in a different file as it's only generating a single table row, but I did it anyway.

**Pull request merged**